Improve README (#43)
cloudposse/bastion: README.md

## Usage

### MFA Providers

This documentation only recognises Google Authenticator and Duo.

> - Google Authenticator

Reviewer: ```suggestion
- **Google Authenticator**
```

  - Google Authenticator is a free, open-source, and easy to setup MFA.
  - It's less secure than Duo because tokens are stored on the server under each user account.

> - Duo

Reviewer: ```suggestion
- **Duo**
```

>   - Duo is an enterprise paid-for service. It has excellent push-notification support so that MFA requests can be approved or declined without opening an app.

Reviewer: ```suggestion
  - Duo is a paid enterprise service. It has excellent mobile push-notification support so that MFA requests can be approved or declined without opening an authenticator app.
```

  - Details here: https://duo.com/pricing

### Running

Refer to the [Environment Variables](#environment-variables) section below to tune how the `bastion` operates.

This command will pull the `latest` tag if it is not already in your image list:
```bash
docker run -p 1234:22 cloudposse/bastion:latest
```

### Building

```bash
git clone https://github.com/cloudposse/bastion.git
cd bastion

> make docker:build

Reviewer: ```suggestion
make docker/build
```

```

### Configuration

#### Recommendations

* Do not allow `root` (or `sudo`) access to this container as doing so would allow remote users to manipulate audit-logs in `/var/log/sudo-io`
* Sync the contents of `/var/log/sudo-io` to a remote, offsite location. If using S3, we recommend enabling bucket-versioning.
* Bind-mount `/etc/passwd`, `/etc/shadow` and `/etc/group` into the container as *read-only*
* Bind-mount `/home` into container; the bastion does not manage authorized keys

#### User Accounts & SSH Keys

> The `bastion` does not attempt to manage user accounts. We suggest using [Cloudposse's github-authorized-keys module](https://github.com/cloudposse/github-authorized-keys/) to automatically provision users; or use the [Helm chart](https://github.com/cloudposse/charts/tree/master/incubator/bastion).

Reviewer: ```suggestion
The `bastion` does not attempt to manage user accounts. We suggest using [our `github-authorized-keys` utility](https://github.com/cloudposse/github-authorized-keys/) to automatically provision users; or use the [Helm chart](https://github.com/cloudposse/charts/tree/master/incubator/bastion).
```
